From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Sun, 6 Sep 2026 23:42:33 +0200
Subject: fix(dracut-systemd): use /sbin/sulogin for
 CONFIG_FEATURE_PREFER_APPLETS

The Dracut module dracut-systemd might call `sulogin -e`, but busybox
does not support the option `-e`. This missing option is reported
upstream at https://github.com/vda-linux/busybox_mirror/issues/36.

The Dracut busybox module skips installing the sulogin applet. This
has no effect when using busybox-static because busybox-static is built
with CONFIG_FEATURE_PREFER_APPLETS enabled. Busybox sh will prefer the
incompatible sulogin applet instead of using the real sulogin binary.

Bug-Debian: https://bugs.debian.org/1146904
Forwarded: https://github.com/dracut-ng/dracut/pull/2722
---
 modules.d/77dracut-systemd/dracut-emergency.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/77dracut-systemd/dracut-emergency.sh b/modules.d/77dracut-systemd/dracut-emergency.sh
index a427a8f..e661c76 100755
--- a/modules.d/77dracut-systemd/dracut-emergency.sh
+++ b/modules.d/77dracut-systemd/dracut-emergency.sh
@@ -36,7 +36,7 @@ if getargbool 1 rd.shell || getarg rd.break; then
 
     if getargbool 0 SYSTEMD_SULOGIN_FORCE; then
         # allows passwordless logins if root account is locked.
-        exec sulogin -e
+        exec /sbin/sulogin -e
     else
         exec sulogin
     fi
